File: src/main/java/my/maryvk/maryvkweb/domain/RelationTypeConverter.java
package my.maryvk.maryvkweb.domain;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter(autoApply = true)
public class RelationTypeConverter implements AttributeConverter<RelationType, Integer> {

    @Override
    public Integer convertToDatabaseColumn(RelationType relationType) {
        return relationType.getId();
    }

    @Override
    public RelationType convertToEntityAttribute(Integer id) {
        return RelationType.get(id);
    }
}

File: src/main/java/my/maryvk/maryvkweb/domain/RelationTypeConverter.kt
package my.maryvk.maryvkweb.domain

import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Converter(autoApply = true)
class RelationTypeConverter : AttributeConverter<RelationType, Int> {

    override fun convertToDatabaseColumn(relationType: RelationType): Int {
        return relationType.id
    }

    override fun convertToEntityAttribute(id: Int): RelationType {
        return RelationType.get(id)
    }
}